examples/helloworld/greeter_server/main.go

```diff
@@ -55,11 +60,25 @@ func (s *server) SayHello(ctx context.Context, in *pb.HelloRequest) (*pb.HelloRe
 }
 
 func main() {
-	lis, err := net.Listen("tcp", port)
-	if err != nil {
-		log.Fatalf("failed to listen: %v", err)
-	}
+	flag.Parse()
+
 	s := grpc.NewServer()
 	pb.RegisterGreeterServer(s, &server{})
-	s.Serve(lis)
+
+	log.Printf("Running hello server on %s ...", *listen)
+	if *httpMode {
+		// Running a gRPC server when you need to integrate with an existing
+		// net/http server on the same port: (using HTTP routing as needed)
+		http.Handle("/", s)
```

Comment:
Understood. In [CockroachDB](https://github.com/cockroachdb/cockroach) we're incrementally migrating from net/rpc to grpc - looks like net/rpc already gets out of the way by using a `DefaultRPCPath` and grpc sets a useful content-type. Seems like that might be enough. Thanks!